TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/workflow/commands/status.py

```diff
@@ -0,0 +1,376 @@
+                ctx.exit(ExitCode.FAILURE.value)
+
+            _display_workflow_status(result)
+        else:
+            # List all workflows with status
+            list_result = service.list_workflows(
+                namespace=namespace,
+                argo_server=argo_server,
+                token=token,
+                insecure=insecure,
+                exclude_completed=not show_all
+            )
+
+            if not list_result['success']:
+                click.echo(f"Error: {list_result['error']}", err=True)
+                ctx.exit(ExitCode.FAILURE.value)
+
+            if list_result['count'] == 0:
+                if show_all:
+                    click.echo(f"No workflows found in namespace {namespace}")
+                    return
+                else:
+                    # No active workflows, try to get the last completed workflow
+                    click.echo(f"No running workflows found in namespace {namespace}")
+
+                    # Try to get completed workflows
+                    completed_result = service.list_workflows(
+                        namespace=namespace,
+                        argo_server=argo_server,
+                        token=token,
+                        insecure=insecure,
+                        exclude_completed=False
+                    )
+
+                    if completed_result['success'] and completed_result['count'] > 0:
+                        # Get the most recent completed workflow by finish time
+                        click.echo("\nShowing last completed workflow:")
+                        click.echo("")
+
+                        # Get status for all workflows to find the most recent by finish time
+                        workflow_statuses = []
+                        for wf_name in completed_result['workflows']:
+                            wf_result = service.get_workflow_status(
+                                workflow_name=wf_name,
+                                namespace=namespace,
+                                argo_server=argo_server,
+                                token=token,
+                                insecure=insecure
+                            )
+                            if wf_result['success'] and wf_result['finished_at']:
+                                workflow_statuses.append(wf_result)
+
+                        # Sort by finished_at timestamp (most recent first)
+                        if workflow_statuses:
+                            workflow_statuses.sort(key=lambda x: x['finished_at'], reverse=True)
+                            _display_workflow_status(workflow_statuses[0])
+                        elif completed_result['workflows']:
+                            # Fallback to first workflow if no finish times available
+                            result = service.get_workflow_status(
+                                workflow_name=completed_result['workflows'][0],
```

Comment:
Are these returned in sorted order?  Will they always be?
The auto-discovery/scanning pattern is making me anxious.  I'd like to see us use something like a label or even a configmap.  We could think about various ways to drive a transaction to keep the current/last workflow easily discoverable.
Can you submit a README file alongside this code that explains the workflow/state machine that we're using to manage workflows.  I want to make sure that we aren't missing anything.
One tool that could be useful would be [argo mutexes](https://argo-workflows.readthedocs.io/en/latest/synchronization/).  We might also want to consider calling the full-migration workflow within another workflow that does some other bookkeeping (including acquiring a mutex).